src/dipdup/datasources/datasource.py

```diff
@@ -71,3 +83,52 @@ def emit_rollback(self, from_level: int, to_level: int) -> None:
 
     def emit_head(self, block: HeadBlockData) -> None:
         super().emit(EventType.head, datasource=self, block=block)
+
+
+@dataclass
+class Subscriptions:
+    address_transactions: Set[str] = Field(default_factory=set)
```

Comment:
Should be grouped by channel instead, for each channel there's a different algorithm for calculating a subscription hash
1. head -> single one
2. blocks -> single one
3. operations -> type! + address? + entrypoint?  # we basically flatten original subscription params
4. big_maps -> ptr? + tag? + contract? + path?

Then the algorithm of adding a new subscriptions will be the following:
1. Flatten subscription args (if applicable)
2. For each subset check if there's an active subscription
3. If not — create a new one